device/coordinate_manager.cpp

```diff
@@ -38,179 +38,53 @@ CoordinateManager::CoordinateManager(
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_physical_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_physical_to_logical;
-        case CoreType::DRAM:
-            return dram_physical_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_physical_to_logical;
-        case CoreType::ARC:
-            return arc_physical_to_logical;
-        case CoreType::PCIE:
-            return pcie_physical_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting physical to logical mapping");
-    }
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_virtual_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_virtual_to_logical;
-        case CoreType::DRAM:
-            return dram_virtual_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_virtual_to_logical;
-        case CoreType::ARC:
-            return arc_virtual_to_logical;
-        case CoreType::PCIE:
-            return pcie_virtual_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting virtual to logical mapping");
-    }
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_translated_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_translated_to_logical;
-        case CoreType::DRAM:
-            return dram_translated_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_translated_to_logical;
-        case CoreType::ARC:
-            return arc_translated_to_logical;
-        case CoreType::PCIE:
-            return pcie_translated_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting translated to logical mapping");
-    }
-}
-
 CoreCoord CoordinateManager::to_physical(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
         case CoordSystem::PHYSICAL:
             return core_coord;
         case CoordSystem::VIRTUAL:
         case CoordSystem::TRANSLATED:
-            return to_physical(to_logical(core_coord));
+        case CoordSystem::LOGICAL: {
+            const tt_xy_pair physical_pair = to_physical_map[core_coord];
+            return CoreCoord(physical_pair.x, physical_pair.y, core_coord.core_type, CoordSystem::PHYSICAL);
+        }
     }
-
-    // Coord system is surely logical.
-    auto& logical_mapping = get_logical_to_physical(core_coord.core_type);
-    return logical_mapping[{core_coord.x, core_coord.y}];
 }
 
 CoreCoord CoordinateManager::to_virtual(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
```

Comment:
Related to what Nigel and Vuk commented on the other function.
If you add an identity map to to_physcal and from_physical, so that it knows how to handle PHYSICAL coords as well, then that fuction (to_physical) will reduce to 2 lines of code. 
But also, in that case, all of these to_virtual, to_logical, to_translated) functions would be simpler, since you wouldn't need to specially handle the case for CoordSystem::PHYSICAL.

And then, even further, we can say that it is fine to "do" the translation even though if same CoreSystem was passed (so, without optimization in case coord system was the same as requested), then all of these functions would also reduce to one lines of code. For example:
```
CoreCoord CoordinateManager::to_virtual(const CoreCoord core_coord) {
  return from_physical_map[{to_physical_map[core_coord], CoordSystem::VIRTUAL}]
}
```

Reply:
Added identity mapping, so no need for to_virtual, to_physical... functions, everything can happen in to function